src/components/TreeListing/TreeListing.tsx

```diff
@@ -49,7 +49,7 @@ export const TreeHeader = (props: { icon?: string; text: string; action?: React.
 export const TreeLeaf = (props: {
   level: number;
   height?: number;
-  icon?: IconName;
+  icon?: IconName | `http://${string}` | `https://${string}`;
```

Comment:
Minor nitpick: due to the way EVEShip.fit is deployed (via https), it will never be possible to have images on http (you can't load images over http when your site is hosted on https for most browsers). So maybe better to just drop the `http://` part, giving an earlier warning to the developer?

Reply:
While this is an component library instead of a website, shouldn't assume people always deploy them on secure contexts (although they definitely should).

Reply:
Yes, this is a component library, but let me reword:

As it is possible this library is used on a https website, it should never contain images to http, as that would fail for those. Having https images for http websites however isn't an issue. In other words: only allowing https means it works for every type of deployment.

That is why I suggest to drop the `http://` part, just to make developers more aware not to use `http`, as that will fail for all sites using this component deployed on `https` :) It is about protecting future-us from making silly mistakes :)